docs/source/conf.py

```diff
@@ -385,27 +387,41 @@ def generate_weights_table(module, table_name, metrics, dataset, include_pattern
     if exclude_patterns is not None:
         weights = [w for w in weights if all(p not in str(w) for p in exclude_patterns)]
 
+    rich_metadata = ["GFLOPs", "Size (MB)"]
+
     metrics_keys, metrics_names = zip(*metrics)
-    column_names = ["Weight"] + list(metrics_names) + ["Params", "Recipe"]
-    column_names = [f"**{name}**" for name in column_names]  # Add bold
+    column_names = ["Weight"] + list(metrics_names) + ["Params"] + rich_metadata + ["Recipe"]  # Final column order
+    column_names_table = [f"**{name}**" for name in column_names if name not in exclude_columns]  # Add bold
+
+    column_patterns = list(
+        zip(
+            column_names,
+            [lambda w: f":class:`{w} <{type(w).__name__}>`"]
+            + [lambda w, metric=m: w.meta["_metrics"][dataset][metric] for m in metrics_keys]
+            + [
+                lambda w: f"{w.meta['num_params']/1e6:.1f}M",
+                lambda w: f"{w.meta['_flops']:.3f}",
+                lambda w: f"{round(w.meta['_weight_size'], 1):.1f}",
+                lambda w: f"`link <{w.meta['recipe']}>`__",
+            ],
```

Comment:
Why is it necessary to convert to lambdas? It brings unnecessary complexity.

Reply:
Somehow this feels unnecessary complex. Can we change the list comprehension to a normal for loop that give us the flexibility to add components on the tuple?